src/main/java/cubicchunks/worldgen/GeneratorPipeline.java

```diff
@@ -102,24 +104,26 @@ public int tick() {
 		// allocate time to each stage depending on busy it is
 		final int sizeCap = 500;
 		int numCubes = 0;
-		for (StageProcessor processor : this.processors) {
-			numCubes += Math.min(sizeCap, processor.processor.getNumInQueue());
+		
+		for (int i = 0; i < this.activeStages; ++i) {
+			numCubes += Math.min(sizeCap, this.stages.get(i).getProcessor().processor.getNumInQueue());
 		}
-		for (StageProcessor processor : this.processors) {
+		
+		for (int i = 0; i < this.activeStages; ++i) {
 			if (numCubes <= 0) {
-				processor.share = 0;
+				this.stages.get(i).getProcessor().share = 0;
 			} else {
-				int size = Math.min(sizeCap, processor.processor.getNumInQueue());
-				processor.share = (float) size/(float) numCubes;
+				int size = Math.min(sizeCap, this.stages.get(i).getProcessor().processor.getNumInQueue());
+				this.stages.get(i).getProcessor().share = (float) size/(float) numCubes;
 			}
 		}
-
+		
 		// process the queues
 		int numProcessed = 0;
-		for (int stage = 0; stage < this.processors.size(); stage++) {
+		for (int i = 0; i < this.activeStages; ++i) {
```

Comment:
This change (stage-->i) is completely pointless, and makes the code less readable.